pkg/sources/adapter/salesforcesource/client/bayeux.go

```diff
@@ -201,6 +243,9 @@ func (b *bayeux) Start(ctx context.Context) error {
 					}
 					b.msgCh <- &crs[i]
 				}
+
+				// process next item.
+				poll <- empty
 			}
```

Comment:
I'm not sure I understand the purpose of this line. If we always re-poll when the loop complete, why can't we just have the whole thing inside a `case default` of the `select`?

Reply:
Falling back to default was the pattern before [Timur's comment.](https://github.com/triggermesh/triggermesh/pull/332#discussion_r768516258)

The default case was incompatible with the backoff. The 3 cases inside the loop are:
- context done, exit function
- error occurred when connecting, wait backoff and do not try to connect
- poll, connect

The error case will wait for the backoff in a go routine and return to the loop so that signaling the context also exit the loop immediately. During backoff we shouldn't try to connect, hence the `select` will wait until the go routine finishes and then it will put a signal at the poll channel.

The poll channel will put a message to re-queue itself if no error occurs during connection.